[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/googlecolab/colabtools/blob/master/notebooks/colab-github-demo.ipynb)


In [ ]:
## print all outputs 
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

# Linear Regression

In [ ]:
#Reading data from CSV
import pandas as pd
df = pd.read_csv("sales.csv")
df.shape
df.head()
df.columns

(3000, 5)

Unnamed: 0  area1_sales  area2_sales  area3_sales  total_sales
0        1000         0.79         0.80         0.02         1.75
1        1001         0.83         0.08         0.18         1.75
2        1002         0.85         0.01         0.22         1.75
3        1003         1.25         0.00         0.03         1.75
4        1004         0.72         0.04         0.10         1.74

Index(['Unnamed: 0', 'area1_sales', 'area2_sales', 'area3_sales',
       'total_sales'],
      dtype='object')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Unnamed: 0   3000 non-null   int64  
 1   area1_sales  3000 non-null   float64
 2   area2_sales  3000 non-null   float64
 3   area3_sales  3000 non-null   float64
 4   total_sales  3000 non-null   float64
dtypes: float64(4), int64(1)
memory usage: 117.3 KB


In [ ]:
#Checking for null values
df.isnull().sum()

Unnamed: 0     0
area1_sales    0
area2_sales    0
area3_sales    0
total_sales    0
dtype: int64

There are no null values in sales data.

In [ ]:
import statsmodels.api as sm
#considerring effects of interactions, if any, to predict the total_sales for a new area using given sales from three areas.
formula = 'total_sales ~ area1_sales + area2_sales+area3_sales+area1_sales*area2_sales*area3_sales'
model = sm.OLS.from_formula(formula = formula, data=df)
result = model.fit()
print(result.summary())

                            OLS Regression Results                            
Dep. Variable:            total_sales   R-squared:                       0.716
Model:                            OLS   Adj. R-squared:                  0.715
Method:                 Least Squares   F-statistic:                     1075.
Date:                Sun, 16 Apr 2023   Prob (F-statistic):               0.00
Time:                        20:37:38   Log-Likelihood:                 830.59
No. Observations:                3000   AIC:                            -1645.
Df Residuals:                    2992   BIC:                            -1597.
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                                          coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------------------
In

In [ ]:
import statsmodels.formula.api as smf
formula = 'total_sales ~ area1_sales + area2_sales+area3_sales+area1_sales*area2_sales*area3_sales'
model = smf.glm(formula = formula, data=df)
result = model.fit()
print("null_deviance:", result.null_deviance)
print()
print("deviance:",result.deviance)

null_deviance: 354.9620154666666

deviance: 100.9639169500172


The p-value for overall model is ~0 hence the model is valid and explains 71% variation in total sales.

The p-value for beta coefficient of interaction between sales of area1 and area2 is 0.029 which is less than 0.05 siginifcance level.  At 5% significance level, we reject null hypothesis. Hence, we conclude that the sales of area1 and area2 interact to affect the total sales for the new area at 5% significance level.

The p-value for beta coefficient of interaction between sales of area1 and area3 is ~0 which is less than any significant alpha, we reject null hypothesis. Hence, we conclude that the sales of area1 and area3 interact to affect the total sales for the new area.

The p-value for beta coefficient of interaction between sales of area2 and area3 is 0.663 which is greater than 0.1 significance level, we fail to reject null hypothesis. We conclude that the sales of area2 and area3 do not interact to affect the total sales for the new area.

The p-value for beta coefficient of interaction between sales of area1, area2 and area3 is ~0 which is less than any significant alpha, we reject null hypothesis. Hence, we conclude that the sales of area1, area2 and area3 interact to affect the total sales for the new area.